org.eclipse.lsp4e/src/org/eclipse/lsp4e/operations/hover/LSPTextHover.java

```diff
@@ -193,28 +194,15 @@ public IRegion getHoverRegion(ITextViewer textViewer, int offset) {
 	private void initiateHoverRequest(@NonNull ITextViewer viewer, int offset) {
 		final IDocument document = viewer.getDocument();
 		this.lastViewer = viewer;
-		this.request = LanguageServiceAccessor
-			.getLanguageServers(document, capabilities -> LSPEclipseUtils.hasCapability(capabilities.getHoverProvider()))
-				.thenApplyAsync(languageServers -> // Async is very important here, otherwise the LS Client thread is in
-													// deadlock and doesn't read bytes from LS
-				languageServers.stream()
-					.map(languageServer -> {
-						try {
-								return languageServer.getTextDocumentService()
-										.hover(LSPEclipseUtils.toHoverParams(offset, document))
-										.get(GET_TIMEOUT_MS, TimeUnit.MILLISECONDS);
-						} catch (ExecutionException | BadLocationException e) {
-							LanguageServerPlugin.logError(e);
-							return null;
-						} catch (InterruptedException e) {
-							LanguageServerPlugin.logError(e);
-							Thread.currentThread().interrupt();
-							return null;
-						} catch (TimeoutException e) {
-							LanguageServerPlugin.logWarning("Could not get hover due to timeout after " + GET_TIMEOUT_MS + " miliseconds", e); //$NON-NLS-1$ //$NON-NLS-2$
-							return null;
-						}
-					}).filter(Objects::nonNull).collect(Collectors.toList()));
+		try {
```

Comment:
To avoid JDT null warnings we need to add
```
if (document == null) {
	return;
}
```
at this point

Reply:
I cannot comment on that line, but to avoid JDT warnings, we also need to annotate the declaration of request to be
`	private CompletableFuture<@NonNull List<@NonNull Hover>> request;`
could you do that?

Reply:
Sure